vscode-dotnet-runtime-library/src/Acquisition/InstallationValidator.ts

```diff
@@ -3,69 +3,209 @@
+     * @returns true if validation passes, false if validation fails. Throws if failOnErr is true and validation fails.
+     * @remarks Validation is not completely exhaustive. Runtime files besides the executable may be missing.
+     * @throws EventBasedError if validation fails and failOnErr is true
+     */
+    public validateDotnetInstall(install: DotnetInstall, dotnetPath: string, validateDirectory = false, failOnErr = true): boolean
+    {
+        const validationContext = {
+            install,
+            dotnetPath,
+            baseErrorMessage: `Validation of .dotnet installation for version ${JSON.stringify(install)} failed:`
+        };
 
-export class InstallationValidator extends IInstallationValidator {
-    public validateDotnetInstall(install: DotnetInstall, dotnetPath: string, isDotnetFolder = false, failOnErr = true): void {
-        const dotnetValidationFailed = `Validation of .dotnet installation for version ${JSON.stringify(install)} failed:`;
-        const folder = path.dirname(dotnetPath);
+        const isValid = validateDirectory
+            ? this.validateDotnetDirectory(validationContext, failOnErr)
+            : this.validateDotnetExecutable(validationContext, failOnErr);
 
-        if(!isDotnetFolder)
+        if (isValid)
         {
-            this.assertOrThrowError(failOnErr, fs.existsSync(folder),
-            `${dotnetValidationFailed} Expected installation folder ${folder} does not exist.`, install, dotnetPath);
+            this.eventStream.post(new DotnetInstallationValidated(install));
+        }
+
+        return isValid;
+    }
 
-            this.assertOrThrowError(failOnErr, fs.existsSync(dotnetPath),
-                `${dotnetValidationFailed} Expected executable does not exist at "${dotnetPath}"`, install, dotnetPath);
+    /**
+     * Validates a .NET executable file by checking:
+     * 1. Parent directory exists
+     * 2. Executable file exists
+     * 3. Path points to a file (not a directory)
+     *
+     * @param context - Validation context containing install details and paths
+     * @param failOnErr - Whether to throw on validation failure or return false
+     * @returns true if all validations pass, false otherwise
+     */
+    private validateDotnetExecutable(context: ValidationContext, failOnErr: boolean): boolean
+    {
+        const { dotnetPath, baseErrorMessage } = context;
+        const parentDirectory = path.dirname(dotnetPath);
 
-            this.assertOrThrowError(failOnErr, fs.lstatSync(dotnetPath).isFile(),
-                `${dotnetValidationFailed} Expected executable file exists but is not a file: "${dotnetPath}"`, install, dotnetPath);
+        // Check if parent directory exists
+        if (!this.validateCondition(
+            fs.existsSync(parentDirectory),
+            `${baseErrorMessage} Expected installation folder ${parentDirectory} does not exist.`,
+            context,
+            failOnErr
+        ))

```

Comment:
nit: I notice multiple if-statements below have this `))` format, which I find a bit weird. I understand it, but it just looks out-of-place. It would probably look a little less weird if both `))` and `{` were on the same line like `)) {` because parens usually signify the end of an expression an squiggly brackets are the start/end of a statement scope (which is why they are separate lines in C#, but not normally in TS, but you have your own formatting for this repo).

Reply:
I agree that's odd. I'll look into seeing if there's something I can do with the auto-formatter to prevent this from happening.